### ISSUE TITLE
[Request] Third party mod installation advice on preview build vs full setup

### ISSUE THREAD

**Issue by Kuiriel** [ZhengPeiRu21/mod-individual-progression#75]
TL;DR: Is it doable to run cmake or otherwise add a module to your preview build, or would it be less time intensive to simply follow the detailed instructions to do a classic install, add each mod you have already provided (including the ah-bot I'm interested in) and then to just copy over your already neatly setup config files?

===

This is so fantastic. Your preview build also resolves a large progression hurdle for someone new to all of the options involved in setting up a classic private server for their family to play on. They're all loving it, so thank you.

Your preview build has just about everything I could want. I note that you mention that people wanting to modify this can set up a classic server and enable all options. However, I'm somewhat concerned about messing up what is already a fantastic balance you have created with a combination of mods and configuration.

If I wanted to install another mod on your existing preview build, is that possible or am I just creating more work in the long run?

I'm under the impression that what you've provided is like a 'compiled' version and not as easily modifiable.

I was looking at installing https://github.com/AyaseCore/mod-ahbot/ but noted that it requires 

* a modules folder (your preview build has a modules folder but that looks like it's for configs only)
* importing SQL (straight forward enough) 
* re-running cmake and launching a 'clean build' of azerothcore.

The last step I'm less sure of - running cmake on an already compiled preview build? It looks like I would just be trying to break things.

Would it be less time intensive to simply follow the detailed instructions to do a classic install, add each mod you have already provided (including the ah-bot I'm interested in) and then to just copy over your already neatly setup config files?

**Comment by bkmangx1** [comment]
Short answer is that you cannot add modules to the preview build.

To add ah-bot you would need to compile your own server with modules of your choice, either with traditional setup or docker. You can follow Azerothcore guides for this. Also make sure you're using [AC fork of Ahbot](https://github.com/azerothcore/mod-ah-bot), because the one you linked is the original port and is very outdated. You are correct about config files, you can reuse the ones provided in preview build. Can also keep the same portable database.

I would recommend joining Azerothcore discord if you have further questions. You will receive answers much quicker there.

**Comment by ZhengPeiRu21** [comment]
I'm apologize for the slow reply - I was not at my computer over a holiday weekend.

I would agree with bkmangx1's suggestions. The preview repack does not support adding additional modules because they are compiled into the server binary, so you will need to set up a build environment, but you can continue to use the same database and configs - you can just replace the server executables in the repack with those produced from your build, and add any new configs to the module configs folder, and it should continue to work with your existing data.
